src/wwwroot/app/services/UserService.js

```diff
@@ -3,28 +3,25 @@ import { first } from 'lodash';
 import { filter } from 'lodash/fp';
 
 const USER_URL = 'user/';
-let _HttpService, _$q, _HttpCacheService, _LoggerService;
-let currentUser = {};
+const userInfo = 'user';
+let _HttpService, _$q, _HttpCacheService, _LoggerService, _StorageService;
 
 export default class UserService {
-   constructor(HttpService, $q, HttpCacheService, LoggerService) {
+   constructor(HttpService, $q, HttpCacheService, LoggerService, LocalStorageService) {
       'ngInject';
       _HttpService      = HttpService;
       _$q               = $q;
       _HttpCacheService = HttpCacheService;
       _LoggerService    = LoggerService;
+      _StorageService   = LocalStorageService;
    }
 
    getUserById(id) {
       return this.getUsers({id}).then(first);
    }
 
    getCurrentUser() {
-      return _$q.when(currentUser);
-   }
-
-   setCurrentUser(user) {
-      currentUser = user;
+      return _StorageService.getValue(userInfo);
```

Comment:
As I see it is possible to store user just in context of some controller or get access to it with session stored data. So I use service, because it is rather possible to get access to its api